xline/src/server/command.rs

```diff
@@ -408,29 +453,19 @@ impl ConflictCheck for Command {
         if (this_req.is_auth_request()) || (other_req.is_auth_request()) {
             return true;
         }
+        let lease_ids_1 = get_lease_ids(this_req);
+        let lease_ids_2 = get_lease_ids(other_req);
 
-        if (this_req.is_lease_request()) && (other_req.is_lease_request()) {
-            #[allow(clippy::wildcard_enum_match_arm)]
-            let lease_id1 = match *this_req {
-                RequestWrapper::LeaseGrantRequest(ref req) => req.id,
-                RequestWrapper::LeaseRevokeRequest(ref req) => req.id,
-                _ => unreachable!("other request can not in this match"),
-            };
-            #[allow(clippy::wildcard_enum_match_arm)]
-            let lease_id2 = match *other_req {
-                RequestWrapper::LeaseGrantRequest(ref req) => req.id,
-                RequestWrapper::LeaseRevokeRequest(ref req) => req.id,
-                _ => unreachable!("other request can not in this match"),
-            };
-            if lease_id1 == lease_id2 {
-                return true;
-            }
-        }
-
-        self.keys()
+        let lease_conflict = lease_ids_1
```

Comment:
It looks like you want to conclude whether the `lease_ids_1` and the `lease_ids_2` are disjointed. We can use  HashSet and is_disjoint to do the same thing.

Reply:
Fixed